mongo/integration/unified/client_entity.go

```diff
@@ -43,6 +47,17 @@ type clientEntity struct {
 }
 
 func newClientEntity(ctx context.Context, em *EntityMap, entityOptions *entityOptions) (*clientEntity, error) {
+	// The "configureFailPoint" command should always be ignored.
+	ignoredCommands := map[string]struct{}{
+		"configureFailPoint": {},
+	}
+	// If not observing sensitive commands, add security-sensitive commands
+	// to ignoredCommands by default.
+	if entityOptions.ObserveSensitiveCommands != nil && !*entityOptions.ObserveSensitiveCommands {
+		for _, cmd := range securitySensitiveCommands {
+			ignoredCommands[cmd] = struct{}{}
+		}
+	}
```

Comment:
We should have been ignoring security-sensitive commands by default with `ignoredCommands`. The new `observeSensitiveCommands` flag explicitly observes the security-sensitive commands we need to examine in the `redacted-commands` tests. Note that this ignoring in the test runner is distinct from the redaction in [`Operation#redactCommand()`](https://github.com/mongodb/mongo-go-driver/blob/master/x/mongo/driver/operation.go#L1411-L1424).

Reply:
That seems right to me. I think only a few drivers _need_ to ignore commands from the connection handshake by adding security-sensitive commands to `ignoredCommands` (or their equivalent) by default.